src/cmr/sizing/core.clj

```diff
@@ -22,32 +22,56 @@
 ;;; Note that all internal operations for estimating size are performed
 ;;; assuming bytes as the units.
 
-(defn get-measurement
+(defn- get-measurement
   [variable]
   (let [dims (get-in variable [:umm :Dimensions])
         total-dimensionality (reduce * (map :Size dims))
         data-type (get-in variable [:umm :DataType])]
     (* total-dimensionality
        (util/data-type->bytes data-type))))
 
+(defn- get-avg-gran-size
+  [vars]
+  (let [value (get-in (first vars) [:umm :SizeEstimation :AvgCompressionRateASCII])]
+    (if (string? value)
+      (read-string value)
+      value)))
+
+(defn- get-avg-compression-rate
+  [vars]
+  (let [value (get-in (first vars) [:umm :SizeEstimation :AverageSizeOfGranulesSampled])]
+    (if (string? value)
+      (read-string value)
+      value)))
+
 (defn estimate-dods-size
   [granule-count variables]
   (let [compression 1
         metadata 0 ; included here for symmetry
         measurements (reduce + (map get-measurement variables))]
     (+ (* granule-count compression measurements) metadata)))
 
-(defn estimate-netcdf3-size
+(defn- estimate-netcdf3-size
   [granule-count variables metadata]
   (let [compression 1
         measurements (reduce + (map get-measurement variables))]
     (+ (* granule-count compression measurements) metadata)))
 
+(defn- estimate-ascii-size
+  [granule-count variables params]
+  (let [metadata 0
+        avg-gran-size (get-avg-gran-size variables)
+        edsc-size-estimate (read-string (:edsc-size-estimate params))
```

Comment:
I don't think we should use the name of one potential client as an input parameter. If I'm reading the wiki page correctly, this should be called something like total-granule-input-bytes (it's the sum of the sizes of the full input granules in the request).

Reply:
@oubiwann

Reply:
Yeah, @chris-durbin is absolutely correct and your original gut instinct was right. My concern about documenting the original source/whose responsible for providing this are better served by a docstring, not a variable name ;-) Let's clarify via example (since we only have one use case/example here) in a docstring where the `total-granule-input-bytes` comes from, and use that as the variable name, per @chris-durbin's suggestion.